Dockerfile

```diff
@@ -1,22 +1,36 @@
+# TODO: make this default to govuk-ruby once it's being pushed somewhere public
 ARG base_image=ruby:2.7.2
-FROM ${base_image}
+ARG app_home=/app
 
-RUN apt-get update -qq && apt-get upgrade -y
-RUN apt-get install -y build-essential nodejs && apt-get clean
-RUN gem install foreman
+FROM ${base_image} AS builder
 
-ENV GOVUK_APP_NAME authenticating-proxy
-ENV PORT 3107
-ENV RAILS_ENV production
+ARG app_home
 
-ENV APP_HOME /app
-RUN mkdir $APP_HOME
+ENV RAILS_ENV=production GOVUK_APP_NAME=authenticating-proxy
 
-WORKDIR $APP_HOME
-ADD Gemfile* $APP_HOME/
-ADD .ruby-version $APP_HOME/
-RUN bundle install
+RUN mkdir -p ${app_home}
 
-ADD . $APP_HOME
+WORKDIR ${app_home}
 
-CMD foreman run web
+COPY Gemfile* .ruby-version ${app_home}/
+
+RUN bundle config set deployment 'true' && \
+    bundle config set without 'development test' && \
+    bundle install -j8 --retry=2
+
+COPY . ${app_home}
+
+
+FROM $base_image
+
```

Comment:
This image should be updated and patched with apt-get update and upgrade

Reply:
I added them again but the packages are only updated only at build time so we will need to rebuild frequently to get the updates.